client-internal/src/main/java/org/terracotta/angela/client/Client.java

```diff
@@ -205,6 +213,12 @@ public void stop() {
 
     logger.info("Killing agent: {} for client:{} instance: {} started from: {}", clientAgentID, clientId, instanceId, parentAgentID);
     final int pid = clientAgentID.getPid();
-    executor.execute(parentAgentID, (IgniteRunnable) () -> AgentController.getInstance().stopClient(instanceId, pid));
+    try {
+      executor.execute(parentAgentID, (IgniteRunnable) () -> AgentController.getInstance().stopClient(instanceId, pid));
+      executor.getGroup().getAllAgents().remove(clientAgentID);   // Dead agent -- remove
```

Comment:
This removal corrects the abrupt test failures in ClientIT -- on Windows, at least, Ignite entered error handing code that called `Runtime.halt()` on the test process if there was a "cluster segmentation" fault observed when shutting down the `IgniteLocalExecutor` -- it didn't like attempting to shutdown a client that no longer existed.

This may not be the "proper" fix -- but it's sufficient.

Reply:
I remember having seen something similar too when refactoring the com layer...